### PR TITLE
Move stream tests to crypto-core

### PR DESCRIPTION
**Pull request by ellisjoe**

### CODE REVIEW

**Comment by ellisjoe** [comment]
@markelliot @uschi2000

File: crypto-core/src/test/java/com/palantir/crypto/cipher/AbstractDecryptingInputStreamTest.java [hunk 1]
```diff
@@ -57,35 +51,26 @@
...
-            FSDataOutputStream fos = fs.create(path);
-            OutputStream os = new FsCipherOutputStream(fos, seekableCipher);
-            os.write(data);
-            os.close();
-            dataWritten = true;
-        }
-
-        FSDataInputStream is = fs.open(path);
-        cis = new FsCipherInputStream(is, seekableCipher);
+        seekableCipher = getSeekableCipher();
```
Comment (ellisjoe) [inline crypto-core/src/test/java/com/palantir/crypto/cipher/AbstractDecryptingInputStreamTest.java]: less than ideal to encrypt before every test, but getSeekableCipher isn't static

File: crypto-core/src/test/java/com/palantir/crypto/io/ByteArraySeekableInput.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * limitations under the License.
+ */
+
+package com.palantir.crypto.io;
+
+import com.palantir.seekio.SeekableInput;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+public final class ByteArraySeekableInput extends ByteArrayInputStream implements SeekableInput {
```
Comment (ellisjoe) [inline crypto-core/src/test/java/com/palantir/crypto/io/ByteArraySeekableInput.java]: move to seek-io?

File: crypto-core/src/test/java/com/palantir/crypto2/io/ByteArraySeekableInput.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * limitations under the License.
+ */
+
+package com.palantir.crypto2.io;
+
+import com.palantir.seekio.SeekableInput;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+public final class ByteArraySeekableInput extends ByteArrayInputStream implements SeekableInput {
```
Comment (markelliot) [inline crypto-core/src/test/java/com/palantir/crypto2/io/ByteArraySeekableInput.java:23]: if you merge the changes over in seek-io can kill this in favor of `InMemorySeekableDataInput`.

**Review by markelliot — APPROVED** [review]

**Pull request merged**